### PR TITLE
<social-link />

### PR DESCRIPTION
**Pull request by vbvnvmfjg**
* [Trello](https://trello.com/c/14sTR80J/58-social-link-component)
* [Preview on page](https://deploy-preview-78--leanwebkit.netlify.com/en)
* [Preview in guide](https://deploy-preview-78--leanwebkit.netlify.com/guide)

### CODE REVIEW

File: dato.config.js [hunk 1]
```diff
@@ -36,7 +36,14 @@ module.exports = (dato, root, i18n) => {
...
 function appSettingsToJson(app) {
-  return pick(app, ['title', 'contact', 'googleAnalyticsTrackingId', 'crispWebsiteId', 'hotjarId'])
+  const { socialLinks } = app
+  const transformSocialLinks = socialLinks.toMap().map(link => {
+    delete link.updatedAt
+    delete link.createdAt
+    return link
+  })
+  const transformApp = Object.assign({}, app, { socialLinks: transformSocialLinks })
+  return pick(transformApp, ['title', 'contact', 'googleAnalyticsTrackingId', 'crispWebsiteId', 'hotjarId', 'socialLinks'])
```
Comment (jbmoelker) [inline dato.config.js]:
```js
function appSettingsToJson(app) {
  const socialLinks = app.socialLinks.toMap().map(item => pick(item, ['platform', 'url']))
  return { socialLinks, ... pick(app, ['title', 'contact', 'googleAnalyticsTrackingId', 'crispWebsiteId', 'hotjarId']) }
}
```

File: src/client/components/app-footer/app-footer.vue [hunk 2]
```diff
@@ -1,37 +1,20 @@
...
-          </a>
-          <a class="app-footer__social-link" v-if="social.linkedin" :href="social.linkedin" @click="track(social.linkedin)">
-            <linkedin-icon />
-            <span class="a11y-sr-only">LinkedIn</span>
-          </a>
-        </div>
+
+      <div class="app-footer__social">
+        <social-link
+          v-for="link in socialLinks"
```
Comment (jbmoelker) [inline src/client/components/app-footer/app-footer.vue]: `in appConfig.socialLinks`

File: src/client/components/app-footer/app-footer.vue [hunk 3]
```diff
@@ -74,9 +57,9 @@ import YoutubeIcon from '../../static/images/youtube.svg'
 
 export default {
   props: ['contentId'],
-  components: { SmartLink, FacebookIcon, TwitterIcon, GooglePlusIcon, InstagramIcon, LinkedinIcon, YoutubeIcon },
+  components: { FacebookIcon, GooglePlusIcon, InstagramIcon, LinkedinIcon, TwitterIcon, SmartLink, SocialLink, YoutubeIcon, },
```
Comment (jbmoelker) [inline src/client/components/app-footer/app-footer.vue]: remove unused icons, also remove imports

File: src/client/components/app-footer/app-footer.vue [hunk 4]
```diff
@@ -74,9 +57,9 @@ import YoutubeIcon from '../../static/images/youtube.svg'
...
   data () {
-    return { appConfig, menuI18n, social }
+    return { appConfig, menuI18n, socialLinks: appConfig.socialLinks }
```
Comment (jbmoelker) [inline src/client/components/app-footer/app-footer.vue]: not needed to put `socialLinks` in as a separate property

File: src/client/components/social-link/social-link.stories.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+import { withReadme } from 'storybook-readme'
+import readme from './readme.md'
+import SocialLink from './'
+
+storiesOf('Social link', module)
+  .addDecorator(withReadme(readme))
+  .addDecorator(withKnobs)
+  .add('Twitter social link', () => ({
+    data: () => ({
+      link: {
```
Comment (jbmoelker) [inline src/client/components/social-link/social-link.stories.js]: only add to a story what you need: `url` and `platform`

File: src/client/components/social-link/social-link.vue [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+import TwitterIcon from '../../static/images/twitter.svg'
+import GooglePlusIcon from '../../static/images/googleplus.svg'
+import InstagramIcon from '../../static/images/instagram.svg'
+import LinkedinIcon from '../../static/images/linkedin.svg'
+import YoutubeIcon from '../../static/images/youtube.svg'
+
+export default {
+  components: { FacebookIcon, TwitterIcon, GooglePlusIcon, InstagramIcon, LinkedinIcon, YoutubeIcon },
+  data () {
+    return { appConfig }
```
Comment (jbmoelker) [inline src/client/components/social-link/social-link.vue]: not used, remove

File: src/client/components/social-link/social-link.vue [hunk 7]
```diff
@@ -0,0 +1,40 @@
+<template>
+  <a
+    :key="link.id"
+    :href="link.url"
+    @click="track(link.url)"
```
Comment (jbmoelker) [inline src/client/components/social-link/social-link.vue]: track is undefined. move here

File: src/client/components/social-link/social-link.vue [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+export default {
+  components: { FacebookIcon, TwitterIcon, GooglePlusIcon, InstagramIcon, LinkedinIcon, YoutubeIcon },
+  data () {
+    return { appConfig }
+  },
+  props: {
+    link: {
+      type: Object,
+      required: true,
```
Comment (jbmoelker) [inline src/client/components/social-link/social-link.vue]: add validator ro change signature to `<social-link platform="" url="">`?

**Review by jbmoelker — CHANGES_REQUESTED** [review]

**Comment by vbvnvmfjg** [comment]
I updated the preview on page/guide in the description.

File: src/client/components/app-footer/app-footer.vue [hunk 9]
```diff
@@ -1,37 +1,24 @@
...
-          <a class="app-footer__social-link" v-if="social.linkedin" :href="social.linkedin" @click="track(social.linkedin)">
-            <linkedin-icon />
-            <span class="a11y-sr-only">LinkedIn</span>
-          </a>
-        </div>
+
+      <ul class="app-footer__social">
+        <li
+          v-for="link in appConfig.socialLinks"
+          :key="link.id"
```
Comment (jbmoelker) [inline src/client/components/app-footer/app-footer.vue:13]: `link.id` is not defined. Either `pick` the IDs in the Dato dump or use `index` here.

**Review by jbmoelker — CHANGES_REQUESTED** [review]
contact form is in this branch, so can't review it on it's own. will move to contact form first.

File: src/client/components/app-footer/app-footer.vue (hunk 9 above)
Comment (vbvnvmfjg) [reply src/client/components/app-footer/app-footer.vue:13]: I picked the `id` in the Dato dump.

**Pull request merged**